### PR TITLE
DatePicker styles

### PR DESCRIPTION
**Pull request by Sorien**
## What does the pull request do?
Changes DatePicker to looks like ComboBox
Adds pseudoclass `:focus` to DatePicker when it's TextBox is focused

## What is the current behavior?

image

## What is the updated/expected behavior with this PR?
image

### CODE REVIEW

**Comment by jmacato** [comment]
@Sorien this looks nice :) ping us when this is done

**Comment by Sorien** [comment]
@jmacato its mostly done just somebody have to decide if its proper way or not couse I'm reusing internal IsFocused property

File: src/Avalonia.Input/InputElement.cs [hunk 1]
```diff
@@ -347,7 +347,7 @@ public Cursor Cursor
         public bool IsFocused
         {
             get { return _isFocused; }
-            private set { SetAndRaise(IsFocusedProperty, ref _isFocused, value); }
+            set { SetAndRaise(IsFocusedProperty, ref _isFocused, value); }
```
Comment (jmacato) [inline src/Avalonia.Input/InputElement.cs]: @Sorien  have you tried making this setter internal then adding an `InternalIsVisibleTo` to whatever internal assembly needs setting this?

**Review by jmacato — CHANGES_REQUESTED** [review]

File: src/Avalonia.Input/InputElement.cs (hunk 1 above)
Comment (Sorien) [reply src/Avalonia.Input/InputElement.cs]: nop, all 3rd party controls that contain any kind of edit (masked edits, tag editors and so on) will need s/t like that so the q is should these controls have their own property like "IsTextBoxInsideFocused" or should we use private IsFocused that is there just for `:focus` pseudoclass

**Comment by kekekeks** [comment]
We probably need separate `IsFocusWithin` property [like WPF does](https://docs.microsoft.com/en-us/dotnet/api/system.windows.uielement.iskeyboardfocuswithin?view=netframework-4.8). This will require several changes to the focus manager that I'd like to make after removing interactive parent shenanigans.

So I'm not sure what to do here, the correct solution is blocked by several major planned changes.

**Comment by grokys** [comment]
I'm going to close this for now, because like @kekekeks said, we should handle this properly rather than hacking in support for it on individual controls. 

@Sorien is there an issue describing this?

**Pull request closed without merging**